php/blocks/class-loader.php

```diff
@@ -250,6 +250,18 @@ public function render_block_template( $block, $attributes ) {
 		$block_lab_attributes = $attributes;
 		$block_lab_config     = $block;
 
+		/**
+		 * Runs in the 'render_callback' of the block.
+		 *
+		 * For example, if a block depends on a JavaScript file,
+		 * this is a good place to call wp_enqueue_script().
+		 * In that case, pass true as the 5th argument ($in_footer) to wp_enqueue_script().
+		 *
+		 * @param array $block The block that is rendered.
+		 * @param array $attributes The block attributes.
+		 */
+		do_action( 'block_lab_render_block_template', $block, $attributes );
```

Comment:
Do you think we should instantiate a `Block` instance here? Maybe at the top of the `render_block_template()` method? Would it be a cleaner way of passing the Block around, instead of an associative array?

It might be outside the scope of this PR, but it would be a backward compatibility break if we decided to do that _after_ releasing this action hook.

Reply:
Yeah, it could help to have a `Block` instance. I don't see a good way to get the block's `ID` that's needed in the [Block constructor](https://github.com/getblocklab/block-lab/blob/c7bce5f2e29e786d4e832192c7cde8e265fd9a0d/php/blocks/class-block.php#L66), though.

This action definitely needs better documentation on how to get the block name, though. Otherwise, they'd have to look through the code a lot.

What do you think about https://github.com/getblocklab/block-lab/pull/340/commits/ea98b4bf08c52db6cf9397776d8746d238551f54?